lib/client.js

```diff
@@ -1117,6 +1118,183 @@ const PreClient = crypto => {
 
       return Promise.all(json.map(IncomingSharingPolicy.decode))
     }
+
+    uploadFileS3(url, checksum, encryptedFile) {
+      // Create the XHR request
+      const xhr = new XMLHttpRequest()
```

Comment:
While XHR is much better than it used to be, it can have x-browser issues. I almost never directly worked with XHR because it was so fraught with inconsistencies (keep in mind these were jQuery days). We have a meeting this morning, let's talk through this.

Reply:
Replaced with fetch.